src/resources/js/routes/Administration.svelte

```diff
@@ -0,0 +1,328 @@
+<script lang="ts">
+  import { Button, Card, Fileupload, Heading, P } from 'flowbite-svelte';
+
+  import { authenticatedUser } from '../stores/apiCredentials';
+  import Nav from '../components/NavBar.svelte';
+  import { translations } from '../stores/localization';
+  import RootServerApi from '../lib/ServerApi';
+  import type { MeetingPartialUpdate } from 'bmlt-server-client';
+
+  let files = $state<FileList | undefined>(undefined);
+  let isLoading = $state(false);
+  let errorMessage = $state<string | null>(null);
+  let isProcessed = $state(false);
+
+  // Statistics tracking.  Stats components are as follows. (worldId is known as 'Committee' in NAWS-speak.)
+  //
+  // totalRows       Total number of rows in the spreadsheet, excluding the header row
+  // malformedRows   Array of row numbers in the spreadsheet of rows missing a bmlt_id or worldId (bmlt_id must be a number; worldId can be '' but not missing altogether)
+  // updated         Array of bmlt_ids for meetings that were successfully updated with a new worldId
+  // noUpdateNeeded  Array of bmlt_ids for meetings whose existing worldId is the same as the new worldId
+  // notedAsDeleted  Array of bmlt_ids for meetings marked as 'deleted' and that are not in the database (i.e. they were actually deleted)
+  // notFound        Array of bmlt_ids of meetings with a new worldId (other than 'deleted') but not found in the database.  This could be because the meeting was deleted from the database but not yet marked as 'deleted' on the spreadsheet, but could be due to an invalid bmlt_id.
+  // errors          Array of strings for meetings for which the server returned an error code.  The string should consist of the bmlt_id and then the error returned by the server.
+  interface Stats {
+    totalRows: number;
+    malformedRows: number[];
+    updated: number[];
+    noUpdateNeeded: number[];
+    notedAsDeleted: number[];
+    notFound: number[];
+    errors: string[];
+  }
+  let stats: Stats = $state({
+    totalRows: 0,
+    malformedRows: [],
+    updated: [],
+    noUpdateNeeded: [],
+    notedAsDeleted: [],
+    notFound: [],
+    errors: []
+  });
+
+  async function processFile(file: File): Promise<void> {
+    try {
+      isLoading = true;
+      errorMessage = null;
+      // Reset stats
+      stats = {
+        totalRows: 0,
+        malformedRows: [],
+        updated: [],
+        noUpdateNeeded: [],
+        notedAsDeleted: [],
+        notFound: [],
+        errors: []
+      };
+
+      console.log('Processing file:', file.name, file.type);
+
+      const extension = getFileExtension(file.name);
+
+      if (!isValidFileType(extension)) {
+        throw new Error('Unsupported file type. Please upload a CSV or Excel file.');
+      }
+
+      const XLSX = await import('xlsx');
+      const buffer = await file.arrayBuffer();
+      const workbook = XLSX.read(buffer, {
+        type: 'array',
+        raw: extension === 'csv'
```

Comment:
this should just be `raw: false`which ensures better type conversion. Otherwise the type check `typeof meetingId !== 'number'` will always fail for csv